bc_obps/reporting/service/report_operation_service.py

```diff
@@ -0,0 +1,116 @@
+from django.db import transaction
+
+from django.forms import model_to_dict
+
+from registration.models.operation import Operation
+from reporting.models import ReportOperationRepresentative, ReportVersion
+from reporting.models.report_operation import ReportOperation
+from service.activity_service import ActivityService
+from service.data_access_service.regulated_product_service import RegulatedProductDataAccessService
+from service.facility_report_service import FacilityReportService
+
+from service.reporting_year_service import ReportingYearService
+
+
+class ReportOperationService:
+    @classmethod
+    def should_show(cls, purpose: str, excluded_purposes: list) -> bool:
+        return purpose not in excluded_purposes
+
+    @classmethod
+    def get_report_operation_data_by_version_id(cls, version_id: int) -> dict:
+        report_operation = cls.get_report_operation_by_version_id(version_id)
+        all_activities = ActivityService.get_all_activities()
+        regulated_products = RegulatedProductDataAccessService.get_regulated_products()
+        reporting_year = ReportingYearService.get_current_reporting_year()
+        purpose = report_operation["registration_purpose"]
+        facility_id = FacilityReportService.get_facility_report_by_version_id(version_id)
+
+        return {
+            "reportOperation": report_operation,
+            "facilityId": facility_id,
+            "allActivities": all_activities,
+            "allRegulatedProducts": regulated_products,
+            "allRepresentatives": report_operation.get("report_operation_representatives", []),
+            "reportType": report_operation.get("operation_report_type"),
+            "showRegulatedProducts": cls.should_show(
+                purpose,
+                [
+                    Operation.Purposes.ELECTRICITY_IMPORT_OPERATION,
+                    Operation.Purposes.REPORTING_OPERATION,
+                    Operation.Purposes.POTENTIAL_REPORTING_OPERATION,
+                ],
+            ),
```

Comment:
the `should_show` method here is introducing confusion, the list comprehensions of python read really well for this:
```py
"showRegulatedProducts": purpose (not) in [a,b,c]
``` 

Otherwise it's more work for the dev to figure out what the call does, especially with a negation inside this function call